package-system/libpng/Findlibpng.cmake

```diff
@@ -0,0 +1,42 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+# 
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+set(MY_NAME "libpng")
+
+set(TARGET_WITH_NAMESPACE "3rdParty::${MY_NAME}")
+
+set(${MY_NAME}_INCLUDE_DIR ${CMAKE_CURRENT_LIST_DIR}/${MY_NAME}/include)
+set(${MY_NAME}_LIBS_DIR ${CMAKE_CURRENT_LIST_DIR}/${MY_NAME}/lib)
+
+if (${PAL_PLATFORM_NAME} STREQUAL "Linux")
+    set(${MY_NAME}_LIBRARY_DEBUG   ${${MY_NAME}_LIBS_DIR}/libpng16.a)
+    set(${MY_NAME}_LIBRARY_RELEASE ${${MY_NAME}_LIBS_DIR}/libpng16.a)
+elseif(${PAL_PLATFORM_NAME} STREQUAL "Android")
+    set(${MY_NAME}_LIBRARY_DEBUG   ${${MY_NAME}_LIBS_DIR}/libpng16.a)
+    set(${MY_NAME}_LIBRARY_RELEASE ${${MY_NAME}_LIBS_DIR}/libpng16.a)
+elseif(${PAL_PLATFORM_NAME} STREQUAL "Windows")
+    set(${MY_NAME}_LIBRARY_DEBUG   ${${MY_NAME}_LIBS_DIR}/libpng16_staticd.lib)
+    set(${MY_NAME}_LIBRARY_RELEASE ${${MY_NAME}_LIBS_DIR}/libpng16_static.lib)
+elseif(${PAL_PLATFORM_NAME} STREQUAL "Mac")
+    set(${MY_NAME}_LIBRARY_DEBUG   ${${MY_NAME}_LIBS_DIR}/libpng16.a)
+    set(${MY_NAME}_LIBRARY_RELEASE ${${MY_NAME}_LIBS_DIR}/libpng16.a)
+elseif(${PAL_PLATFORM_NAME} STREQUAL "iOS")
+    set(${MY_NAME}_LIBRARY_DEBUG   ${${MY_NAME}_LIBS_DIR}/libpng16.a)
+    set(${MY_NAME}_LIBRARY_RELEASE ${${MY_NAME}_LIBS_DIR}/libpng16.a)
+endif()
```

Comment:
Could this be changed to just 
```
if(${PAL_PLATFORM_NAME} STREQUAL "Windows")
    set(${MY_NAME}_LIBRARY_DEBUG   ${${MY_NAME}_LIBS_DIR}/libpng16_staticd.lib)
    set(${MY_NAME}_LIBRARY_RELEASE ${${MY_NAME}_LIBS_DIR}/libpng16_static.lib)
else()
    set(${MY_NAME}_LIBRARY_DEBUG   ${${MY_NAME}_LIBS_DIR}/libpng16.a)
    set(${MY_NAME}_LIBRARY_RELEASE ${${MY_NAME}_LIBS_DIR}/libpng16.a)
endif()
```

Reply:
Maybe, but I was following the pattern shown in other "Find" files in the package system.